pkg/provisioner/resourcehandlers.go

```diff
@@ -176,6 +225,106 @@ func createConfigMap(obc *v1alpha1.ObjectBucketClaim, ep *v1alpha1.Endpoint, lab
+			log.Error(err, "probably not fatal, retrying")
+			return false, nil
+		}
+		// secret already exists
+		return true, nil
+	})
+	return secret, err
+}
+
+func getExistingConfigMap(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) (configMap *corev1.ConfigMap, err error) {
+	err = wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
+		configMap, err = c.CoreV1().ConfigMaps(obc.Namespace).Get(context.TODO(), configMapNameForClaim(obc), metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Not found is valid here which returns no error and a nil configMap
+				configMap = nil
+				err = nil
+				return true, nil
+			}
+			// for other errors that are likely intermittent, give the opportunity to retry until timeout
+			log.Error(err, "probably not fatal, retrying")
+			return false, nil
+		}
+		// configMap already exists
+		return true, nil
+	})
+	return configMap, err
+}
+
+func deleteSecret(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) error {
+	deleteFunc := func() error {
+		return c.CoreV1().Secrets(obc.Namespace).Delete(context.TODO(), secretNameForClaim(obc), metav1.DeleteOptions{})
+	}
+	getFunc := func() error {
+		_, err := c.CoreV1().Secrets(obc.Namespace).Get(context.TODO(), secretNameForClaim(obc), metav1.GetOptions{})
+		return err
+	}
+	logD.Info("deleting secret", "secret name", secretNameForClaim(obc))
+	return deleteResourceAndWait(deleteFunc, getFunc, retryInterval, retryTimeout)
+}
+
+func deleteConfigMap(obc *v1alpha1.ObjectBucketClaim, c kubernetes.Interface, retryInterval, retryTimeout time.Duration) error {
+	deleteFunc := func() error {
+		return c.CoreV1().ConfigMaps(obc.Namespace).Delete(context.TODO(), configMapNameForClaim(obc), metav1.DeleteOptions{})
+	}
+	getFunc := func() error {
+		_, err := c.CoreV1().ConfigMaps(obc.Namespace).Get(context.TODO(), configMapNameForClaim(obc), metav1.GetOptions{})
+		return err
+	}
+	logD.Info("deleting configmap", "cm name", configMapNameForClaim(obc))
+	return deleteResourceAndWait(deleteFunc, getFunc, retryInterval, retryTimeout)
+}
+
+func deleteResourceAndWait(deleteFunc func() error, getFunc func() error, retryInterval, retryTimeout time.Duration) error {
+	var err error
+	err = wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
+		err = deleteFunc()
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Already deleted
+				err = nil
+				return true, nil
+			}
+			// for other errors that are likely intermittent, give the opportunity to retry until timeout
+			log.Error(err, "probably not fatal, retrying")
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	err = wait.PollImmediate(retryInterval, retryTimeout, func() (done bool, err error) {
+		err = getFunc()
```

Comment:
Is the point of the get func to wait in case the resource has a finalizer? Otherwise, since you are not returning the resource in cases of Delete failure, I don't see the value in get. Also, a top func comment would help here.

Reply:
The `Get` waits for the resource to be deleted. A new resource can't be created unless the old one is already gone.